Userland/Libraries/LibRegex/RegexParser.h

```diff
@@ -197,6 +201,17 @@ class ECMA262Parser final : public Parser {
     bool parse_capture_group(ByteCode&, size_t&, bool unicode, bool named);
     Optional<CharClass> parse_character_class_escape(bool& out_inverse, bool expect_backslash = false);
     bool parse_nonempty_class_ranges(Vector<CompareTypeAndValuePair>&, bool unicode);
+
+    // Used only by B.1.4, Regular Expression Patterns (Extended for use in browsers)
+    bool parse_quantifiable_assertion(ByteCode&, size_t&, bool named);
+    bool parse_extended_atom(ByteCode&, size_t&, bool named);
+    bool parse_inner_disjunction(ByteCode& bytecode_stack, size_t& length, bool unicode, bool named);
+    bool parse_invalid_braced_quantifier(); // Note: This function either parses and *fails*, or doesn't parse anything and returns false.
+    bool parse_legacy_octal_escape_sequence(ByteCode& bytecode_stack, size_t& length);
+    Optional<u8> parse_legacy_octal_escape();
+
+    // Let's keep them separate, for now?
+    bool m_should_use_browser_extended_grammar { false };
```

Comment:
Yes, definitely. What I'd like to do, and I might have mentioned this on IRC before, is adding an Annex B flag to the JS parser and engine, so you can conditionally enable/disable all the legacy stuff - I'd say for now just always enable this and I'll come back to it, I'm sure there are other Annex B features already built in unconditionally.

The browser would have that always on, and the JS repl by default with an option to disable, or something like that.

Reply:
Then I guess we can always pass the `Internal_BrowserExtended` flag from LibJS here, and any further changes can be done in LibJS.